lib/moab/file_group_difference.rb

```diff
@@ -263,14 +263,13 @@ def tabulate_renamed_files(matching_signatures, basis_signature_hash, other_sign
           fid.basis_path = basis_only_paths[n]
           fid.other_path = other_only_paths[n]
           fid.signatures << signature
-          case true
-            when fid.basis_path.nil?
-              fid.change = 'copyadded'
-              fid.basis_path = basis_paths[0]
-            when fid.other_path.nil?
-              fid.change = 'copydeleted'
-            else
-              fid.change = 'renamed'
+          if fid.basis_path.nil?
+            fid.change = 'copyadded'
+            fid.basis_path = basis_paths[0]
+          elsif fid.other_path.nil?
+            fid.change = 'copydeleted'
+          else
+            fid.change = 'renamed'
           end
```

Comment:
Wondering whether the `case` is used here to explicitly allow it to fall-through from one `when` to the next and so on; a cascading flow of checking/correcting values for `fid.change`?

Reply:
ruby doesn't do fall-through on case statements:
https://docs.ruby-lang.org/en/2.4.0/syntax/control_expressions_rdoc.html#label-case+Expression
https://stackoverflow.com/questions/31443922/does-rubys-case-statement-fall-through
https://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-case
https://www.ruby-forum.com/topic/98827

(docs don't say it explicitly, but they do imply a lack of fall-through, and the internet comments seem to confirm this)